libcloud/compute/types.py

```diff
@@ -35,7 +35,27 @@
 ]
 
 
-class Provider(object):
+class Type(object):
+    @classmethod
+    def tostring(cls, value):
+        """Return the string representation of the state object attribute
+        :param str value: the state object to turn into string
+        :return: the uppercase string that represents the state object
+        :rtype: str
+        """
+        return value.upper()
```

Comment:
I think I'm fine with this change, but since it's a class method it might also also be a good idea to throw if users passes in an invalid enum value.

In any case, current approach works for now.